docs/src/user_docs/assembly/execution_contexts.md

```diff
@@ -0,0 +1,105 @@
+## Execution contexts
+Miden assembly program execution can span multiple isolated contexts. An execution context defines its own memory space which is not accessible from other execution contexts.
+
+All programs start executing in a *root* context. Thus, the main procedure of a program is always executed in the root context. To move execution into a different context, we can invoke a procedure using the `call` instruction. In fact, any time we invoke a procedure using the `call` instruction, the procedure is executed in a new context. We refer to all non-root contexts as *user contexts*.
+
+While executing in a user context, we can request some procedures to be executed in the root context. This can be done via the `syscall` instruction. The set of procedures which can be invoked via the `syscall` instruction is limited by a [kernel](#kernels) against which a program is compiled. Once the procedure called via `syscall` returns, the execution moves back to the user context from which it was invoked. The diagram below illustrates this graphically:
```

Comment:
nit: I think "...we can request to execute some procedures in the root context." would be preferable.

also nit: "... is limited by the kernel..." instead of "... is limited by a kernel..."